sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubBufferedProducerClient.cs

```diff
@@ -1216,18 +1205,44 @@ public virtual async ValueTask DisposeAsync()
         ///   This method will modify class-level state and should be synchronized.  It is assumed that callers hold responsibility for
         ///   ensuring synchronization concerns; this method should be invoked after any primitives have been acquired.
         ///
-        ///   Callers are also assumed to own responsibility for any validation that the client is in the intended state before calling.
+        ///   Callers are also assumed to own responsibility for stopping all publishing and performing any necessary
+        ///   validation of client state before calling.
         /// </remarks>
         ///
-        internal virtual Task ClearInternalAsync(CancellationToken cancellationToken = default)
+        internal virtual void ClearInternal(CancellationToken cancellationToken = default)
         {
-            // ======================================================================
-            //  NOTE:
-            //    This method is currently just a stub; full functionality will be
-            //    added in a later set of changes.
-            // ======================================================================
+            var operationId = GenerateOperationId();
+
+            try
+            {
+                Logger.BufferedProducerClearStart(Identifier, EventHubName, operationId);
+                cancellationToken.ThrowIfCancellationRequested();
```

Comment:
nit: could we move this to the top of the foreach and eliminate the one below at the bottom of it?

Reply:
Yes, yes we could.   That would also eliminate the need to poll the token in the loop as well.

Reply:
Cool - IIRC, there was one other place with this same pattern.

Reply:
Yeah, `FlushInternalAsync` had similar.   I cleaned that up as well.